### PR TITLE
fix #4837 Update glob due to vulnerability in dep

### LINKED ISSUE #4837: mocha 6.2.3 minimatch fixed version causing security scans to fail
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend that you _not_ install Mocha globally.

### Description

minimatch package versions before 3.0.5 are vulnerable to Regular Expression Denial of Service (ReDoS). Fixed version of minimatch (3.0.4) for mocha version 6.2.3 is causing cloud computing scans to fail. 

In the past I've seen doing some upgrade for security reasons to older major versions so I wanted to know if I need to upgrade this service that is in maintenance mode or not. Thanks a lot in advance :) 

### Steps to Reproduce

N/A

**Expected behavior:** Security scans don't fail.

**Actual behavior:** N/A

**Reproduces how often:** 100%

### Versions

- The output of `mocha --version` and `node_modules/.bin/mocha --version`: 6.2.3

### PR DESCRIPTION
**Pull request by jb2311**
### Description of the Change
The minimatch npm package version <=3.0.4 has a security vulnerability explained here: https://github.com/advisories/GHSA-f8q6-p94x-37v3

Mocha has the [glob package 7.2.0 as a dependency](https://github.com/mochajs/mocha/blob/master/package.json#L64), which in turn has [minimatch ^3.0.4 as a dependency](https://github.com/isaacs/node-glob/blob/v7.2.0/package.json#L23)

This results in a high security risk warning for consumers of mocha. See https://github.com/mochajs/mocha/issues/4937 for an example.

This PR changes the glob dependency to 8.1.0 which no longer depends on the insecure version of minimatch.

### CODE REVIEW

**Comment by jb2311** [comment]
@juergba this should be ready to be tested again. 2 things I have concerns about. #1 The tests referenced coffee-script, with a hyphen, but when I ran the tests, I got a "module not found error" and it looks like the hyphen package is depreciated, and mocha's package.json references "coffeescript" with no hyphen, so I updated those tests. Additionally, I got an error during linting that   "eslint-plugin-n" is missing. I though it was my local setup, but it also happened when the tests ran in the CI process. Anyways, I added that package to the dev deps. Let me know if you have any questions or concerns

**Comment by juergba** [comment]
@jb2311 thank you for this PR.

We will see wether our CI tests will pass. Tbh I don't trust your `package-lock.json`.

**Comment by jb2311** [comment]
esline-plugin-node is no longer being maintained, so eslint-plugin-n (https://github.com/eslint-community/eslint-plugin-n) is a fork that is still active. However, I have no idea why it's erroring out without it suddenly. It happened on the first test run, when I did nothing but update the glob package https://github.com/mochajs/mocha/actions/runs/4368808387/jobs/7681675449

File: package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "diff": "5.0.0",
     "escape-string-regexp": "4.0.0",
     "find-up": "5.0.0",
-    "glob": "7.2.0",
+    "glob": "8.1.0",
```
Comment (juergba) [inline package.json:64]: glob has published v9*. Is mocha affected by v9 breaking changes, or why are you limiting your upgrade to v8?

File: package.json [hunk 2]
```diff
@@ -167,7 +167,7 @@
     "./lib/nodejs/file-unloader.js": false,
     "./lib/nodejs/parallel-buffered-runner.js": false,
     "./lib/nodejs/serializer.js": false,
-    "./lib/nodejs/worker.js": false,
+    ".  /lib/nodejs/worker.js": false,
```
Comment (juergba) [inline package.json:170]: please remove space.

File: package.json [hunk 3]
```diff
@@ -183,4 +183,4 @@
       "pre-commit": "lint-staged"
     }
   }
-}
+}
```
Comment (juergba) [inline package.json:186]: is there an empty line missing at the end?

**Review by juergba — COMMENTED** [review]
I'm not convinced completely, I'm afraid there might be some unwanted side-effects by `windowsPathsNoEscape`. Anyway let's do it.

**Comment by jb2311** [comment]
@juergba updated

File: package.json [hunk 4]
```diff
@@ -184,3 +184,4 @@
     }
   }
 }
+
```
Comment (juergba) [inline package.json:187]: Please ... I don't want to mess with linters or whatever, so please just leave this untouched. There has to be one empty line at the end. There is no need to change on this eof for this PR.

**Comment by jb2311** [comment]
@juergba okay, I think the whitespace should be appropriate.

File: package.json (hunk 4 above)
Comment (jb2311) [reply package.json:187]: It looks like the file in master doesnt have the empty line at the end

**Pull request merged**